app/src/main/java/org/breezyweather/common/basic/GeoActivity.kt

```diff
@@ -53,12 +53,6 @@ abstract class GeoActivity : AppCompatActivity() {
         BreezyWeather.instance.setTopActivity(this)
     }
 
-    @Deprecated("Deprecated in Java")
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        super.onActivityResult(requestCode, resultCode, data)
-        BreezyWeather.instance.setTopActivity(this)
```

Comment:
Can't the top activity be missing in some cases if we are not setting it here?

Reply:
I tested this by logging the calls of `onActivityResult` and it was only called when finishing the SearchActivity which is handled via the new [ActivityForResultContract](https://github.com/breezy-weather/breezy-weather/blob/dad2f11f9a274ee3c42decebc5df741852e6d5f6/app/src/main/java/org/breezyweather/main/MainActivity.kt#L125-L135) instead.

Other cases should be covered by [onNewIntent](https://github.com/breezy-weather/breezy-weather/blob/dad2f11f9a274ee3c42decebc5df741852e6d5f6/app/src/main/java/org/breezyweather/common/basic/GeoActivity.kt#L51) and [onResume](https://github.com/breezy-weather/breezy-weather/blob/dad2f11f9a274ee3c42decebc5df741852e6d5f6/app/src/main/java/org/breezyweather/common/basic/GeoActivity.kt#L57) where `setTopActivity` is also called.